client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java

```diff
@@ -920,18 +918,13 @@ public void testQueryMetrics() throws Exception {
         }
         response = client.query(insertStmtBuilder.toString(), settings).get();
 
-        serverStats = response.getOperationStatistics().serverStatistics;
-        System.out.println("Server stats: " + serverStats);
-        System.out.println("Client stats: " + stats.clientStatistics);
-
-        // Server stats: ServerStatistics{"numRowsRead"=10, "numRowsWritten"=10, "totalRowsToRead"=0, "numBytesRead"=651, "numBytesWritten"=651}
-        Assert.assertTrue(serverStats.numBytesRead > 0);
-        Assert.assertTrue(serverStats.numBytesWritten > 0);
-        Assert.assertEquals(serverStats.numRowsRead, rowsToInsert); // 10 rows in the table
-        Assert.assertEquals(serverStats.numRowsWritten, rowsToInsert); // 10 rows inserted
-        Assert.assertEquals(serverStats.totalRowsToRead, 0);
-        Assert.assertEquals(serverStats.resultRows, rowsToInsert);
-        Assert.assertTrue(stats.clientStatistics.getElapsedTime("query") > 0);
+        metrics = response.getMetrics();
+        System.out.println("Server read rows: " + metrics.getMetric(ServerMetrics.NUM_ROWS_READ).getLong());
+        System.out.println("Client stats: " + metrics.getMetric(ClientMetrics.OP_DURATION).getLong());
+
+        Assert.assertEquals(metrics.getMetric(ServerMetrics.NUM_ROWS_READ).getLong(), rowsToInsert); // 10 rows in the table

```

Comment:
We discuss to have getter methods (explicit) in this case, i need `metrics.getMetric(ServerMetrics.NUM_ROWS_READ).getLong())` why not just metric.getNumRowsRead() return long

Reply:
There are going to be many metrics over time. Getting only few of them is not a major usecase. 
I think users will have list of metrics what they want to publish to some metrics storage - so such getter is more useful than getter for each metric.